v2/kotlin-sdk/app/src/main/java/com/mongodb/app/ItemActivity.kt

```diff
@@ -0,0 +1,482 @@
+    private lateinit var airplaneModeItem: MenuItem
+    private lateinit var filterSwitch: SwitchMaterial
+    private lateinit var fab: View
+
+    private var connectivityEnabled = true
+    private var syncDisabledIcon: Drawable? = null
+    private var syncEnabledIcon: Drawable? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_item)
+
+        val toolbar = findViewById<View>(R.id.item_menu) as Toolbar
+        setSupportActionBar(toolbar)
+
+        fab = findViewById<View>(R.id.floating_action_button)
+        fab.setOnClickListener { onFabClicked() }
+
+        initializeResources()
+
+        filterSwitch = findViewById(R.id.filter_switch)
+
+        recyclerView = findViewById(R.id.item_list)
+        recyclerView.layoutManager = LinearLayoutManager(this)
+        recyclerView.setHasFixedSize(true)
+        recyclerView.addItemDecoration(DividerItemDecoration(this, DividerItemDecoration.VERTICAL))
+    }
+
+    /**
+     *  On start, open a realm that contains items for the current user.
+     *  Query the realm for 'Item' objects, sorted by a stable order that remains
+     *  consistent between runs, and display the collection using a recyclerView.
+     */
+    override fun onStart() {
+        super.onStart()
+        val user = realmApp.currentUser
+        if (user == null) {
+            startActivity(Intent(this, LoginActivity::class.java))
+        } else {
+            config = SyncConfiguration.Builder(user, setOf(Item::class))
+                .initialSubscriptions { realm ->
+                    add(
+                        realm.query<Item>("owner_id == $0", realmApp.currentUser!!.id),
+                        "User's Items"
+                    )
+                }
+                .waitForInitialRemoteData()
+                .build()
+            this.realm = Realm.open(config)
+            CoroutineScope(Dispatchers.IO).launch {
+                realm.subscriptions.waitForSynchronization()
+            }
+            val query = realm.query<Item>()
+            itemAdapter = ItemAdapter(query.find(), realm, query.asFlow())
+            recyclerView.adapter = itemAdapter
+
+            filterSwitch.setOnCheckedChangeListener { _, isChecked ->
+                // Disallow adding tasks when only filtering by completed tasks
+//                fab.isEnabled = !isChecked
+
+                CoroutineScope(Dispatchers.IO).launch {
+                    realm.subscriptions.update {
+                        val queryMine = realm.query<Item>("owner_id == $0", realmApp.currentUser!!.id)
+                        val queryMineCompleted = realm.query<Item>("owner_id == $0 && isComplete == $1", realmApp.currentUser!!.id, true)
+
+                        if (isChecked) {
+                            removeAll()
+                            queryMineCompleted.subscribe(SUBSCRIPTION_MINE_COMPLETED)
```

Comment:
It was decided to have the subscription change between "my" tasks and "everyone's" tasks. The "completed" status could be demonstrated with a simple filter, rather than a subscription. I'm ok with this change happening in a subsequent PR.

Reply:
Yeah, this is just the "legacy" XML version which isn't used anymore.